### PR TITLE
fix: Send updated selector on streaming reconnect

### PR DESCRIPTION
**Pull request by keelerm84**
BEGIN_COMMIT_OVERRIDE
chore: Send updated selector on streaming reconnect
END_COMMIT_OVERRIDE

### CODE REVIEW

**Comment by keelerm84** [comment]
This relies on a pending eventsource change, but at least this gets the PR up and ready for review.

File: internal/datasourcev2/streaming_data_source.go [hunk 1]
```diff
@@ -304,11 +304,8 @@ func (sp *StreamProcessor) consumeStream(stream *es.Stream, closeWhenReady chan<
 	}
 }
 
-func (sp *StreamProcessor) subscribe(closeWhenReady chan<- struct{}, selector fdv2proto.Selector) {
+func (sp *StreamProcessor) subscribe(closeWhenReady chan<- struct{}, _ fdv2proto.Selector) {
```
Comment (tanderson-ld) [inline internal/datasourcev2/streaming_data_source.go:307]: Do you think this param will ever be used agian?

File: subsystems/data_destination.go [hunk 2]
```diff
@@ -11,6 +11,9 @@ import (
 // Do not use it.
 // You have been warned.
 type DataDestination interface {
+	// Selector returns the last known selector for the data store.
```
Comment (tanderson-ld) [inline subsystems/data_destination.go]: Consider adding that the selector may be unknown.  Is that the `fdv2proto.NoSelector()`?

File: subsystems/data_destination.go [hunk 3]
```diff
@@ -11,6 +11,9 @@ import (
 // Do not use it.
```
Comment (tanderson-ld) [inline subsystems/data_destination.go:13]: Update comment to remove use of "sink".

File: internal/datasourcev2/streaming_data_source.go (hunk 1 above)
Comment (keelerm84) [reply internal/datasourcev2/streaming_data_source.go:307]: Given our latest discussion, I wouldn't think so. But I don't want to remove it in this PR just yet as it's going to balloon up. How about I remove it in a separate PR?

File: subsystems/data_destination.go (hunk 2 above)
Comment (keelerm84) [reply subsystems/data_destination.go]: Done.

File: subsystems/data_destination.go (hunk 3 above)
Comment (keelerm84) [reply subsystems/data_destination.go:13]: Do you have a recommendation for different wording? I'm not sure what I would say differently here.

File: internal/datasourcev2/streaming_data_source.go (hunk 1 above)
Comment (keelerm84) [reply internal/datasourcev2/streaming_data_source.go:307]: https://github.com/launchdarkly/go-server-sdk/pull/228

File: subsystems/data_destination.go (hunk 3 above)
Comment (keelerm84) [reply subsystems/data_destination.go:13]: Ryan had a good suggestion. How does that sound?

**Review by kinyoklion — APPROVED** [review]

**Review by tanderson-ld — APPROVED** [review]

**Review by tanderson-ld — APPROVED** [review]

**Pull request merged**